src/test/scala/it/pagopa/interop/agreementprocess/AgreementUpdateSpec.scala

```diff
@@ -0,0 +1,68 @@
+package it.pagopa.interop.agreementprocess
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import it.pagopa.interop.agreementmanagement.client.model.UpgradeAgreementSeed
+import org.scalatest.matchers.should.Matchers._
+import org.scalatest.wordspec.AnyWordSpecLike
+import it.pagopa.interop.authorizationmanagement.client.model.{
+  ClientAgreementAndEServiceDetailsUpdate,
+  ClientComponentState
+}
+
+import java.util.UUID
+
+class AgreementUpdateSpec extends AnyWordSpecLike with SpecHelper with ScalatestRouteTest {
+
+  import agreementApiMarshaller._
+
+  "Agreement update" should {
+    "succedd" in {
+      val newerDescriptor   = SpecData.publishedDescriptor.copy(version = "10")
+      val currentDescriptor = SpecData.deprecatedDescriptor.copy(version = "1")
+      val eService          = SpecData.eService.copy(descriptors = Seq(newerDescriptor, currentDescriptor))
+      val consumer          = SpecData.tenant.copy(id = requesterOrgId)
+      val agreement         =
+        SpecData.activeAgreement.copy(
+          eserviceId = eService.id,
+          descriptorId = currentDescriptor.id,
+          consumerId = consumer.id
+        )
+      val newAgreement      =
+        agreement.copy(
+          id = UUID.randomUUID(),
+          eserviceId = eService.id,
+          descriptorId = newerDescriptor.id,
+          consumerId = consumer.id
+        )
+
+      val seed = UpgradeAgreementSeed(descriptorId = newerDescriptor.id, SpecData.defaultStamp.get)
+
+      mockAgreementRetrieve(agreement)
+      mockEServiceRetrieve(eService.id, eService)
+      mockAgreementUpgrade(agreement.id, seed, newAgreement)
```

Comment:
```suggestion
      mockAgreementUpdate(agreement.id, seed, newAgreement)
```
and this should not return a new agreement, but the same agreement with the updated fields